### PR TITLE
upgrade to typescript 4, upgrade eslint and add esModuleInterop true

### LINKED ISSUE #15: Enable `esModuleInterop` by default in TS configs
"Is there a good reason not to use esModuleInterop in TypeScript?"

In @dagda1 we trust 

I would set it in your base tsconfig
I always set it
I think import * was the original intention but after babel, it never really caught on
I find import * as a default import confusing

### PR DESCRIPTION
**Pull request by dagda1**
We want to set `"esModuleInterop": true` for all packages to avoid the `import * from 'blah';` ambiguity.

This PR also

- upgrades typescript to version 4
- updates eslint
- ~~adds `ts-node` and `@frontside/eslint-config` as direct dependencies of `@frontside/typescript`~~

### CODE REVIEW

**Comment by cowboyd** [comment]
> @dagda1 adds ts-node and @frontside/eslint-config as direct dependencies of @frontside/typescript

- We need to confirm that we can invoke the `ts-node` executable within a monorepo for both the case where it is hoisted to the root as well as when it is in `packages/foo/node_modules/@frontside/typescrcript/node_modules/ts-node
- There is a whole [rigamarole][1] `@frontside/eslint-config` and the way it resolves plugins that we need to slog through:

[1]: https://github.com/thefrontside/javascript/issues/10

File: packages/typescript/package.json [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
   "description": "TypeScript Development Tools at Frontside",
   "author": "Frontside Engineering <engineering@frontside.com>",
   "dependencies": {
-    "@frontside/tsconfig": "1.0.0",
-    "eslint": "4.18.2",
-    "typescript": "^3.9.7"
+    "@frontside/eslint-config": "1.2.0",
+    "@frontside/tsconfig": "1.1.0",
+    "@typescript-eslint/eslint-plugin": "^3.10.1",
+    "@typescript-eslint/parser": "^3.10.1",
```
Comment (minkimcello) [inline packages/typescript/package.json]: As Charles said, extending `frontside/eslint-config` is a whole different thing. We were working on a workaround by creating custom commands with `frontside/typescript`. So for now you can omit `@frontside/eslint-config`, `@typescript-eslint/eslint-plugin`, and `@typescript-eslint/parser`.

File: packages/typescript/package.json [hunk 2]
```diff
@@ -1,14 +1,21 @@
...
+    "eslint": "7.7.0",
+    "ts-node": "^9.0.0",
```
Comment (minkimcello) [inline packages/typescript/package.json]: Until we can confirm that `ts-node` is executable in a monorepo, should we put this off for another PR?

**Review by minkimcello — CHANGES_REQUESTED** [review]

File: packages/typescript/package.json (hunk 2 above)
Comment (dagda1) [reply packages/typescript/package.json]: I'm actually not a fan of ts-node getting resolved this way.  I've had problems with trying to do this in the past for other things so happy to remove.

For me, the important things are

- upgrade typescript to 4.0
- esModuleInterop: true is in the base config

Comment (dagda1) [reply packages/typescript/package.json]: removed ts-node

File: packages/typescript/package.json (hunk 1 above)
Comment (dagda1) [reply packages/typescript/package.json]: removed

**Review by minkimcello — APPROVED** [review]

**Comment by jnicklas** [comment]
Should we merge this?

**Comment by dagda1** [comment]
@jnicklas I would like to, to get rid of those `import *` imports and typescript 4 has no breaking changes.  I did an upgrade in another contract yesterday.  

The Typescript maintainers don't do semver for some weird reason.

@cowboyd @minkimcello @taras  ??

**Comment by cowboyd** [comment]
I say go for it

**Comment by dagda1** [comment]
@cowboyd the deed is done!

**Comment by dagda1** [comment]
@cowboyd  shall I upgrade bigtest?
